src/store/swapOrderSlice.ts

```diff
@@ -0,0 +1,206 @@
+  listenerMiddleware.startListening({
+    actionCreator: slice.actions.setSwapOrder,
+    effect: (action, listenerApi) => {
+      const { dispatch } = listenerApi
+      const swapOrder = action.payload
+      const oldStatus = selectSwapOrderStatus(listenerApi.getOriginalState(), swapOrder.orderUid)
+      const newStatus = swapOrder.status
+
+      if (oldStatus === newStatus || newStatus === undefined) {
+        return
+      }
+
+      switch (newStatus) {
+        case 'created':
+          const safeInfo = selectSafeInfo(listenerApi.getState())
+
+          dispatch(
+            showNotification({
+              title: 'Order created',
+              message:
+                safeInfo.data?.threshold === 1
+                  ? 'Waiting for the transaction to be executed'
+                  : 'Waiting for confirmation from signers of your Safe',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+
+          break
+        case 'presignaturePending':
+          dispatch(
+            showNotification({
+              title: 'Order waiting for signature',
+              message: 'Waiting for confirmation from signers of your Safe',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+          break
+        case 'open':
+          dispatch(
+            showNotification({
+              title: 'Order transaction confirmed',
+              message: 'Waiting for order execution by the CoW Protocol',
+              groupKey,
+              variant: 'info',
+            }),
+          )
+          break
+        case 'fulfilled':
+          dispatch(slice.actions.deleteSwapOrder(swapOrder.orderUid))
+          if (oldStatus === undefined) {
+            return
+          }
+          dispatch(
+            showNotification({
+              title: 'Order executed',
+              message: 'Your order has been successful',
+              groupKey,
+              variant: 'success',
+            }),
+          )
+          break
+        case 'expired':
+          dispatch(slice.actions.deleteSwapOrder(swapOrder.orderUid))
+          if (oldStatus === undefined) {
+            return
+          }
+          dispatch(
+            showNotification({
+              title: 'Order expired',
+              message: 'Your order has reached the expiry time and has become invalid',
+              groupKey,
+              variant: 'warning',
+            }),
+          )
+          break
+        case 'cancelled':
+          dispatch(slice.actions.deleteSwapOrder(swapOrder.orderUid))
+          if (oldStatus === undefined) {
+            return
+          }
+          dispatch(
+            showNotification({
+              title: 'Order cancelled',
+              message: 'Your order has been cancelled',
+              groupKey,
+              variant: 'warning',
+            }),
+          )
+          break
+      }
```

Comment:
This can be made much more declaritive. E.g. a config where keys are order statuses and values are messages.

Reply:
It's a bit more complicated, because we have some states that we are not reacting too and also in some of the cases we are dispatching multiple events, so I would leave it as it is for now.